src/Sylius/Bundle/ApiBundle/ApiPlatform/Factory/MergingExtractorResourceMetadataFactory.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Extractor\ExtractorInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Sylius\Bundle\ApiBundle\ApiPlatform\ResourceMetadataPropertyValueResolver;
+
+/**
+ * @experimental
+ * This class is overwriting ApiPlatform ExtractorResourceMetadataFactory to allow yaml files to be merged into api platform config
+ */
+final class MergingExtractorResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    /** @var ExtractorInterface */
+    private $extractor;
+
+    /** @var ResourceMetadataFactoryInterface */
+    private $decorated;
+
+    /** @var ResourceMetadataPropertyValueResolver */
+    private $resourceMetadataPropertyValueResolver;
+
+    /** @var array */
+    private $defaults;
+
+    /** @var array */
+    private $resources = ['description', 'iri', 'itemOperations', 'collectionOperations', 'graphql'];
```

Comment:
```suggestion
    private const RESOURCES = ['description', 'iri', 'itemOperations', 'collectionOperations', 'graphql'];
```

however, can the `RESOURCES` name be more meaningful? Or perhaps, it is just me who does not understand this naming?